kv/service.go

```diff
@@ -24,20 +24,25 @@ type Service struct {
 	Logger *zap.Logger
 	Config ServiceConfig
 
-	IDGenerator    influxdb.IDGenerator
+	IDGenerator influxdb.IDGenerator
+
+	// special ID generator that never returns bytes with backslash,
+	// comma, or space. Used to support very specific encoding of org &
+	// bucket into the old measurement in storage.
+	OrgBucketIDs influxdb.IDGenerator
+
 	TokenGenerator influxdb.TokenGenerator
 	influxdb.TimeGenerator
 	Hash Crypt
-
-	// Organization & bucket specific validation.
-	IsValidOrgBucketID func(influxdb.ID) bool
 }
 
 // NewService returns an instance of a Service.
 func NewService(kv Store, configs ...ServiceConfig) *Service {
 	s := &Service{
-		Logger:         zap.NewNop(),
-		IDGenerator:    snowflake.NewIDGenerator(),
+		Logger:      zap.NewNop(),
+		IDGenerator: snowflake.NewIDGenerator(),
+		// Seed the random number generator with the current time
+		OrgBucketIDs:   rand.NewOrgBucketID(time.Now().UnixNano()),
```

Comment:
Perhaps you want to provide an option to set the seed? It will make testing easier because the ids generated will be deterministically generated.